packages/oats-runtime/test/make.spec.ts

```diff
@@ -20,6 +21,15 @@ describe('createMakerWith', () => {
     expect(result).toBeInstanceOf(TestClass);
     expect(result.a).toEqual(['a']);
   });
+
+  it('when additional properties is false then remove undefined values', () => {
```

Comment:
1) we can test just the makeObject call, which is I think *way* simpler as we do not need to define a TestClass
2) I think we should remove undefined when additionalProperties in the openapi spec is some schema or 'false' -- but not otherwise.

Reply:
1. ok thanks for the tip, I will update the test.
2. would this be a good place to filter out the undefined values? 🤔 
```
      const propResult: Make<any> = additionalProp(value[index], opts);
      if (propResult.isError()) {
        return propResult.errorPath(index);
      } else if (value[index] === undefined) {
        // when a schema is provided as additional props we should also skip the undefined values
        continue;
      }
```

Reply:
mmm won't let me get away with just makeObject, it does not accept undefined as value
<img width="1135" alt="image" src="https://user-images.githubusercontent.com/9282200/222141532-5cbc1124-d1d2-44e1-af89-3061183401c6.png">